test/common/http/conn_manager_impl_test_base.h

```diff
@@ -29,6 +29,13 @@ using testing::NiceMock;
 namespace Envoy {
 namespace Http {
 
+struct SetupOpts {
```

Comment:
Nice refactor. Not a blocker but you could consider use a builder kind of pattern so that setup can just become:

```
setup(SetupOpt().setFoo(nonDefaultValue))
```